cmd/cluster-agent-cloudfoundry/app/app.go

```diff
@@ -0,0 +1,261 @@
+	}
+
+	// get hostname
+	hostname, err := util.GetHostname()
+	if err != nil {
+		return log.Errorf("Error while getting hostname, exiting: %v", err)
+	}
+	log.Infof("Hostname is: %s", hostname)
+
+	// setup the forwarder
+	keysPerDomain, err := config.GetMultipleEndpoints()
+	if err != nil {
+		log.Error("Misconfiguration of agent endpoints: ", err)
+	}
+	f := forwarder.NewDefaultForwarder(keysPerDomain)
+	f.Start()
+	s := serializer.NewSerializer(f)
+
+	aggregatorInstance := aggregator.InitAggregator(s, nil, hostname, "cluster_agent")
+	aggregatorInstance.AddAgentStartupTelemetry(fmt.Sprintf("%s - Datadog Cluster Agent", version.AgentVersion))
+
+	log.Infof("Datadog Cluster Agent is now running.")
+
+	// Setup a channel to catch OS signals
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, os.Interrupt, syscall.SIGTERM)
+
+	// initialize BBS Cache before starting provider/listener
+	err = initializeBBSCache()
+	if err != nil {
+		return err
+	}
+
+	// create and setup the Autoconfig instance
+	common.SetupAutoConfig(config.Datadog.GetString("confd_path"))
+	// start the autoconfig, this will immediately run any configured check
+	common.StartAutoConfig()
+
+	var clusterCheckHandler *clusterchecks.Handler
+	clusterCheckHandler, err = setupClusterCheck(mainCtx)
+	if err != nil {
+		log.Errorf("Error while setting up cluster check Autodiscovery %v", err)
+	}
+
+	// Start the cmd HTTPS server
+	// We always need to start it, even with nil clusterCheckHandler
+	// as it's also used to perform the agent commands (e.g. agent status)
+	sc := clusteragent.ServerContext{
+		ClusterCheckHandler: clusterCheckHandler,
+	}
+	if err = api.StartServer(sc); err != nil {
+		return log.Errorf("Error while starting agent API, exiting: %v", err)
+	}
+
+	// Block here until we receive the interrupt signal
+	<-signalCh
+
+	// retrieve the agent health before stopping the components
+	// GetStatusNonBlocking has a 100ms timeout to avoid blocking
+	health, err := health.GetStatusNonBlocking()
+	if err != nil {
+		log.Warnf("Cluster Agent health unknown: %s", err)
+	} else if len(health.Unhealthy) > 0 {
+		log.Warnf("Some components were unhealthy: %v", health.Unhealthy)
+	}
+
+	// Cancel the main context to stop components
+	mainCtxCancel()
+
+	log.Info("See ya!")
+	log.Flush()
+	return nil
+}
+
+func initializeBBSCache() error {
+	pollInterval := time.Second * time.Duration(config.Datadog.GetInt("cloud_foundry_bbs.poll_interval"))
+	// NOTE: we can't use GetPollInterval in ConfigureGlobalBBSCache, as that causes import cycle
+	bc, err := cloudfoundry.ConfigureGlobalBBSCache(
+		config.Datadog.GetString("cloud_foundry_bbs.url"),
+		config.Datadog.GetString("cloud_foundry_bbs.ca_file"),
+		config.Datadog.GetString("cloud_foundry_bbs.cert_file"),
+		config.Datadog.GetString("cloud_foundry_bbs.key_file"),
+		pollInterval,
+		false,
+	)
+	if err != nil {
+		return fmt.Errorf("Failed to initialize BBS Cache: %s", err.Error())
```

Comment:
`fmt.Errorf(` usually, the go linter don't accept an message starting with an uppercase. I may be wrong tho.

Reply:
The linter did pass, but you're right, I'll make it lowercase.

Reply:
Hmm, OTOH there's bunch of other `fmt.Errorf` errors throughout the codebase that start with a capital letter and the linter doesn't seem to be complaining, so I guess I'll just leave it as it is 🤷‍♂

Reply:
ok let's adapt to whatever practices are used in this codebase indeed.

Reply:
Still let's be consistent everywhere then.